common/compression/installshield_cab.cpp

```diff
@@ -99,41 +100,71 @@ class InstallShieldCabinet : public Archive {
 	FileMap _map;
 	String _baseName;
 	Common::Array<VolumeHeader> _volumeHeaders;
+    bool _useSearchMan;
 
 	static bool readVolumeHeader(SeekableReadStream *volumeStream, VolumeHeader &inVolumeHeader);
 
 	String getHeaderName() const;
 	String getVolumeName(uint volume) const;
 };
 
-InstallShieldCabinet::InstallShieldCabinet() : _version(0) {
+InstallShieldCabinet::InstallShieldCabinet() : _version(0), _useSearchMan(false) {
 }
 
-bool InstallShieldCabinet::open(const String &baseName) {
-	// Store the base name so we can generate file names
-	_baseName = baseName;
+bool InstallShieldCabinet::open(const String *baseName, const FSNode *node) {
+    // Store the base name so we can generate file names
+    if (baseName) {
+        _baseName = *baseName;
+        _useSearchMan = true;
+    } else if (node) {
+        _baseName = node->getPath();
+        _useSearchMan = false;
+    } else {
+        return false;
+    }
+
+    if (_baseName.hasSuffix(".cab") || _baseName.hasSuffix(".hdr")) {
+        _baseName.erase(_baseName.size() - 5, String::npos);
```

Comment:
`- 5` or still `- 4`? I think the last letter could get erased with this.

Reply:
Definitely - 5. IS cabs follow a naming scheme of <name>#.cab, where # is a digit that gets incremented by 1 with every new file. We only need to store the <name> part, sans that digit, and then we re-add it as needed.